modules/openapi-generator/src/main/resources/go-server/routers.mustache

```diff
@@ -108,7 +108,11 @@ func EncodeJSONResponse(i interface{}, status *int,{{#addResponseHeaders}} heade
 		w.WriteHeader(http.StatusOK)
 	}
 
-	return json.NewEncoder(w).Encode(i)
+    if i != nil {
+        return json.NewEncoder(w).Encode(i)
+    }
+
+    return nil
```

Comment:
FYI. Filed and merged https://github.com/OpenAPITools/openapi-generator/pull/14740 to replace 4-space with tabs